langchain-core/src/callbacks/base.ts

```diff
@@ -197,7 +198,7 @@ abstract class BaseCallbackHandlerMethodsClass {
    * Called at the end of a Tool run, with the tool output and the run ID.
    */
   handleToolEnd?(
-    output: string,
+    output: string | ToolMessage,
```

Comment:
I think this should be an any type -- a tool should be allowed to return anything, and we shouldn't be limiting the return type in anyway
```suggestion
    output: any,
```

Reply:
How would we handle things that aren't a string or tool message? Type check and e.g. JSON.stringify?